File: chazm-model/src/test/java/runtimemodels/chazm/model/OrganizationTest.java
package runtimemodels.chazm.model;

import com.google.inject.Guice;
import com.google.inject.Injector;
import com.google.inject.Provider;
import org.junit.jupiter.api.Test;
import runtimemodels.chazm.api.Organization;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertAll;

public class OrganizationTest {

    private final Injector injector = Guice.createInjector(new OrganizationModule());
    private final Provider<Organization> provider = injector.getProvider(Organization.class);

    @Test
    public void testOrganization() {
        final Organization o1 = provider.get();
        final Organization o2 = provider.get();

        assertAll(
                () -> assertThat(o1).isNotNull(),
                () -> assertThat(o2).isNotNull(),
                () -> assertThat(o1).isNotSameAs(o2),
                () -> assertThat(o1.getAgents().size()).isEqualTo(0),
                () -> assertThat(o1.getAttributes().size()).isEqualTo(0),
                () -> assertThat(o1.getCapabilities().size()).isEqualTo(0),
                () -> assertThat(o1.getCharacteristics().size()).isEqualTo(0),
                () -> assertThat(o1.getInstanceGoals().size()).isEqualTo(0),
                () -> assertThat(o1.getPmfs().size()).isEqualTo(0),
                () -> assertThat(o1.getPolicies().size()).isEqualTo(0),
                () -> assertThat(o1.getRoles().size()).isEqualTo(0),
                () -> assertThat(o1.getSpecificationGoals().size()).isEqualTo(0)
        );
    }

}

File: chazm-model/src/test/kotlin/runtimemodels/chazm/model/OrganizationTest.kt
package runtimemodels.chazm.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertAll
import org.koin.standalone.StandAloneContext.loadKoinModules
import org.koin.standalone.StandAloneContext.stopKoin
import org.koin.standalone.get
import org.koin.test.KoinTest
import runtimemodels.chazm.api.organization.Organization
import runtimemodels.chazm.model.koin.OrganizationModules

internal class OrganizationTest : KoinTest {
    @Test
    fun `the initial state of a new instance`() {
        val organization: Organization = get()

        assertAll(
            { assertThat(organization).isNotNull },
            { assertThat(organization.agents.size).isEqualTo(0) },
            { assertThat(organization.attributes.size).isEqualTo(0) },
            { assertThat(organization.capabilities.size).isEqualTo(0) },
            { assertThat(organization.characteristics.size).isEqualTo(0) },
            { assertThat(organization.instanceGoals.size).isEqualTo(0) },
            { assertThat(organization.pmfs.size).isEqualTo(0) },
            { assertThat(organization.policies.size).isEqualTo(0) },
            { assertThat(organization.roles.size).isEqualTo(0) },
            { assertThat(organization.specificationGoals.size).isEqualTo(0) }
        )
    }

    @Test
    fun `no two instances are the same`() {
        val organization: Organization = get()

        assertAll(
            { assertThat(organization).isNotSameAs(get<Organization>()) },
            { assertThat(organization).isNotSameAs(get<Organization>()) }
        )
    }

    companion object {
        @BeforeAll
        @JvmStatic
        fun beforeAll() {
            loadKoinModules(OrganizationModules)
        }

        @AfterAll
        @JvmStatic
        fun afterAll() {
            stopKoin()
        }
    }
}
